tests/test_synced_collection.py

```diff
@@ -602,3 +672,112 @@ def synced_list(self):
+
+    def store(self, data):
+        dataset = zarr.group(self._store).require_dataset(
+            'test', overwrite=True, shape=1, dtype='object', object_codec=numcodecs.JSON())
+        dataset[0] = data
+
+
+@pytest.mark.skipif(not ZARR, reason='test requires the zarr package')
+class TestZarrList(TestJSONList):
+
+    @pytest.fixture(autouse=True)
+    def synced_list(self):
+        self._tmp_dir = TemporaryDirectory(prefix='zarrlist_')
+        self._store = zarr.DirectoryStore(self._tmp_dir.name)
+        self._name = 'test'
+        yield ZarrList(name=self._name, store=self._store)
+        self._tmp_dir.cleanup()
+
+    def store(self, data):
+        dataset = zarr.group(self._store).require_dataset(
+            'test', overwrite=True, shape=1, dtype='object', object_codec=numcodecs.JSON())
+        dataset[0] = data
+
+
+@pytest.mark.skipif(not REDIS, reason='test requires the redis package and running redis-server')
+class TestRedisDict(TestJSONDict):
+
+    @pytest.fixture
+    def synced_dict(self, request):
+        self._client = RedisClient
+        request.addfinalizer(self._client.flushall)
+        self._name = 'test'
+        yield RedisDict(name=self._name, client=self._client)
+
+    def store(self, data):
+        self._client.set(self._name, json.dumps(data).encode())
+
+
+@pytest.mark.skipif(not REDIS, reason='test requires the redis package and running redis-server')
+class TestRedisList(TestJSONList):
+
+    @pytest.fixture
+    def synced_list(self, request):
+        self._client = RedisClient
+        request.addfinalizer(self._client.flushall)
+        self._name = 'test'
+        yield RedisList(name=self._name, client=self._client)
+
+    def store(self, data):
+        self._client.set(self._name, json.dumps(data).encode())
+
+
+@pytest.mark.skipif(not PYMONGO, reason='test requires the pymongo package and mongodb server')
+class TestMongoDict(TestJSONDict):
+
+    @pytest.fixture
+    def synced_dict(self, request):
+        self._client = MongoClient
+        self._name = 'test'
+        yield MongoDict(name=self._name, client=self._client,
+                        database='test_db', collection='test_dict')
+        self._client.test_db.test_dict.drop()
+
+    def store(self, data):
+        data_to_insert = {'MongoDict::name': self._name, 'data': data}
+        self._client.test_db.test_dict.replace_one({'MongoDict::name': self._name}, data_to_insert)
+
+    def test_write_invalid_type(self, synced_dict, testdata):
+        # mongodict return InvalidDocument error for objects
+        class Foo(object):
+            pass
+
+        key = 'write_invalid_type'
+        synced_dict[key] = testdata
+        assert len(synced_dict) == 1
+        assert synced_dict[key] == testdata
+        d2 = Foo()
+        with pytest.raises(bson.errors.InvalidDocument):
+            synced_dict[key + '2'] = d2
+        assert len(synced_dict) == 1
+        assert synced_dict[key] == testdata
```

Comment:
Please catch the exception in the backend implementation and raise the correct exception type so that we can remove this backend-specific test.

Reply:
Okay, will add that.